code/game/objects/effects/effect_system/effect_system.dm

```diff
@@ -70,7 +72,8 @@ would spawn and follow the beaker, even if it is carried or thrown.
 	for(var/j in 1 to steps_amt)
 		sleep(5)
 		step(E,direction)
-	addtimer(CALLBACK(src, .proc/decrement_total_effect), 20)
+	if(!QDELETED(src))
```

Comment:
It's questionable if the elements should keep stepping when the source is deleted. Left as is for now.